skimage/transform/pyramids.py

```diff
@@ -3,16 +3,17 @@
 from scipy import ndimage as ndi
 from ..transform import resize
 from ..util import img_as_float
+from ._warps import _multichannel_default
 
 
-def _smooth(image, sigma, mode, cval):
+def _smooth(image, sigma, mode, cval, multichannel=None):
     """Return image with each channel smoothed by the Gaussian filter."""
-
+    multichannel = _multichannel_default(multichannel, image.ndim)
     smoothed = np.empty(image.shape, dtype=np.double)
 
-    # apply Gaussian filter to all dimensions independently
-    if image.ndim == 3:
-        for dim in range(image.shape[2]):
+    # apply Gaussian filter to all channels independently
+    if multichannel:
+        for dim in range(image.shape[-1]):
```

Comment:
This seems unnecessarily complex — isn't it equivalent to run a single gaussian filter with the corresponding sigma = 0?